src/Storages/ObjectStorage/DataLakes/DataLakeConfiguration.h

```diff
@@ -222,6 +230,19 @@ class DataLakeConfiguration : public BaseStorageConfiguration, public std::enabl
                 /* table_engine_definition */nullptr
             );
         }
+        if ((*settings)[DataLakeStorageSetting::iceberg_catalog_type].value == DatabaseDataLakeCatalogType::ICEBERG_REST)
```

Comment:
A bit confusing that we use iceberg related staff in a common class `DataLakeConfiguration` which is not supposed to know about concrete data lakes. Should these settings be even prefixed with `iceberg_`? I understand we support writes only for iceberg at the moment, but the same catalogs can be used with delta, and so they will have the same settings set.